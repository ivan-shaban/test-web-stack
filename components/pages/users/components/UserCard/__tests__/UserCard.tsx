import React from 'react'
import {
    UserCard,
    Props,
} from '../UserCard'
import renderer from 'react-test-renderer'
import {mount} from 'enzyme'

const defaultProps: Props = {
    user: {
        address: 'address',
        description: 'description',
        id: 0,
        dob: new Date('1990-01-01'),
        name: 'name',
        image: 'image',
        createdAt: new Date('1999-01-01'),
        updatedAt: new Date('1999-02-01'),
    },
    onClick: jest.fn(),
}

const createTree = (props?: Partial<Props>) => {
    return renderer.create(<UserCard {...defaultProps} {...props}/>).toJSON()
}

const createWrapper = (props?: Partial<Props>) => {
    return mount(<UserCard {...defaultProps} {...props}/>)
}

describe('snapshots', () => {
    it('default', () => {
        const tree = createTree()
        expect(tree).toMatchSnapshot()
    })

    it('with disabled state', () => {
        const tree = createTree({isDisabled: true})
        expect(tree).toMatchSnapshot()
    })
})

it('should handle onClick', function () {
    const onClick = jest.fn()
    const wrapper = createWrapper({onClick})
    wrapper.simulate('click')

    expect(onClick).toHaveBeenCalledWith(defaultProps.user)
})
