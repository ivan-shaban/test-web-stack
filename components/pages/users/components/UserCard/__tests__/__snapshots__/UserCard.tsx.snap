// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots default 1`] = `
<div
  className="base"
  onClick={[Function]}
>
  <img
    alt="edit icon"
    className="editIcon"
    draggable="false"
    src="/static/icons/edit-icon.svg"
  />
  <img
    alt="name's avatar"
    className="avatar"
    draggable="false"
    src="image"
  />
  <div
    className="credentials"
  >
    <p
      className="title"
    >
      name
    </p>
    <p
      className="createdTitle"
    >
      created 
      <span
        className="createdValue"
      >
        01.01.1990
      </span>
    </p>
  </div>
  <p
    className="description"
  >
    description
  </p>
</div>
`;

exports[`snapshots with disabled state 1`] = `
<div
  className="base base__disabled"
  onClick={[Function]}
>
  <img
    alt="edit icon"
    className="editIcon"
    draggable="false"
    src="/static/icons/edit-icon.svg"
  />
  <img
    alt="name's avatar"
    className="avatar"
    draggable="false"
    src="image"
  />
  <div
    className="credentials"
  >
    <p
      className="title"
    >
      name
    </p>
    <p
      className="createdTitle"
    >
      created 
      <span
        className="createdValue"
      >
        01.01.1990
      </span>
    </p>
  </div>
  <p
    className="description"
  >
    description
  </p>
</div>
`;
